cmd/target-browser/cmd/backup.go

```diff
@@ -42,14 +44,40 @@ for specific backupPlan using available flags and options.`,
 
   # List of backups: filter by [tvkInstanceUID]
   kubectl tvk-target-browser get backup --tvk-instance-uid <uid> --target-name <name> --target-namespace <namespace>
-  
-  # List of backups: filter by [expiry-date] (NOT SUPPORTED)
-  kubectl tvk-target-browser get backup --backup-plan-uid <uid> --expiry-date <date> --target-name <name> --target-namespace <namespace>
 
-  # List of backups: filter by [creation-date] (NOT SUPPORTED)
-  kubectl tvk-target-browser get backup --backup-plan-uid <uid> --creation-date <date> --target-name <name> --target-namespace <namespace>
+  # List of backups: filter by [expirationStartDate] and [expirationEndDate]
+  kubectl tvk-target-browser get backup --expiration-start-date <expiration-start-date> --expiration-end-date <expiration-end-date>--target-name <name> --target-namespace <namespace>
+
+  # List of backups: filter by [creationStartDate] and [creationEndDate]
+  kubectl tvk-target-browser get backup --creation-start-date <creation-start-date> --creation-end-date <creation-end-date>--target-name <name> --target-namespace <namespace>
 `,
+
 	RunE: func(cmd *cobra.Command, args []string) error {
+		if cmd.Flags().Changed(ExpirationStartDateFlag) && expirationEndDate == "" ||
+			cmd.Flags().Changed(ExpirationEndDateFlag) && expirationEndDate == "" {
+			return fmt.Errorf("[%s] flag value cannot be empty", ExpirationEndDateFlag)
+		}
+		var ts *time.Time
+		var err error
+		if expirationEndDate != "" {
+			ts, err = parseTimestamp(expirationEndDate)
+			if err != nil {
+				return fmt.Errorf("[%s] flag invalid value. Usage - %s", ExpirationEndDateFlag, expirationEndDateUsage)
+			}
+			expirationEndDate = ts.Format(time.RFC3339)
```

Comment:
is format validity check not required with `time.Parse` func.?

Reply:
here `ts` is valid timestamp
No, Any valid timestamp can be formatted into RFC3339.